src/lightning_app/utilities/app_logs.py

```diff
@@ -109,18 +143,20 @@ def _app_logs_reader(
         th.start()
 
     # Print logs from queue when log event is available
-    user_log_start = "<<< BEGIN USER_RUN_FLOW SECTION >>>"
-    start_timestamp = None
+    component_logs_started = {component_name: False for component_name in component_names}
 
-    # Print logs from queue when log event is available
     try:
         while True:
-            log_event = read_queue.get(timeout=None if follow else 1.0)
-            if user_log_start in log_event.message:
-                start_timestamp = log_event.timestamp + timedelta(seconds=0.5)
+            log_event: _LogEvent = read_queue.get(timeout=None if follow else 1.0)
 
-            if start_timestamp and log_event.timestamp > start_timestamp:
+            if component_logs_started[log_event.component_name] and not is_internal_log(log_event.message):
                 yield log_event
+                continue
+
+            section = LogSection.USER_RUN_FLOW if log_event.component_name == "flow" else LogSection.USER_RUN_WORK
+
+            if log_delimiter_begin_for(section) in log_event.message:
+                component_logs_started[log_event.component_name] = True
```

Comment:
If the app is restarted, will the logs contain the `delimiter_begin` multiple times ?

Reply:
Right now, `lightning show logs` doesn't behave the same as the UI, which clears the logs after every app restart. We could do this in the CLI as well by passing `start` query parameter to the logs endpoint. We could include that in this PR or in a follow up.

Reply:
I think the logs should be cleared after every app stop/start, let's do it in a separate PR or this one if you prefer.

Maybe with an additional flag, we could show all logs in the future :D 
`lightning show logs --all`

Reply:
I agree that the logs should get cleared after every app start/stop. can we add that to this PR